surefire-api/src/main/java/org/apache/maven/surefire/api/util/DefaultRunOrderCalculator.java

```diff
@@ -55,7 +55,7 @@ public DefaultRunOrderCalculator( RunOrderParameters runOrderParameters, int thr
         this.runOrder = runOrderParameters.getRunOrder();
         this.sortOrder = this.runOrder.length > 0 ? getSortOrderComparator( this.runOrder[0] ) : null;
         Long runOrderRandomSeed = runOrderParameters.getRunOrderRandomSeed();
-        this.random = new Random( runOrderRandomSeed == null ? System.nanoTime() : runOrderRandomSeed );
+        random = runOrderRandomSeed == null ? null : new Random( runOrderRandomSeed );
```

Comment:
Why change in this way ...?
When `rundom` will be null we will have NPE from `Collections.shuffle( testClasses, random );`

Reply:
IMHO here `runOrderParameters.getRunOrderRandomSeed();` never be null